docs/source/user_guide/installation/install-from-source.rst

```diff
@@ -179,25 +197,62 @@ and run the tests to check your installation.
 Manual install
 ~~~~~~~~~~~~~~
 
-From the ``PyBaMM/`` directory, you can install PyBaMM using
+We recommend to install PyBaMM within a virtual environment, in order
+not to alter any distribution Python files.
+
+To create a virtual environment ``env`` within your current directory type:
+
+.. code:: bash
+
+   virtualenv env
+
+You can then “activate” the environment using:
+
+.. code:: bash
+
+   source env/bin/activate
+
+Now all the calls to pip described below will install PyBaMM and its
+dependencies into the environment ``env``. When you are ready to exit
+the environment and go back to your original system, just type:
+
+.. code:: bash
+
+   deactivate
+
+From the ``PyBaMM/`` directory inside the virtual environment, you can install PyBaMM using
 
 .. code:: bash
 
 	  pip install .
 
+If you want to install PyBaMM in an "editable mode", use the following command:
+
+.. code:: bash
+
+	  pip install -e .
+
 If you intend to contribute to the development of PyBaMM, it is convenient to
 install in "editable mode", along with all the optional dependencies and useful
-tools for development and documentation:
+tools for development and documentation.
+
+Due to the ``--no-build-isolation`` flag in the "editable mode" command, you first need to install the build-time dependencies inside the virtual environment:
+
+.. code:: bash
+
+	pip install scikit-build-core==0.10.3 casadi cmake pybind11
```

Comment:
Recappulating the above conversations:
1. editable rebuilds are experimental. Hence, version pinning is important to avoid breaking changes.
2. We can entirely remove editable-rebuilds and have normal editable installations. This would remove the need to pin the versions